### PR TITLE
Change choice of q in qary-lattice generation

### PR DESCRIPTION
**Pull request by ludopulles**
Let random lattice generation use the methods changed in fplll/fplll#526.
Note: adds support for values of 'q' above 2^31 when calling IntegerMatrix.random with methods 'ntrulike', 'ntrulike2' and 'qary'.

Note: G6K does not need any modification.

Resolves: post by @malb in https://github.com/fplll/fplll/issues/526#issuecomment-2166150270

### CODE REVIEW

File: tests/test_gso.py [hunk 1]
```diff
@@ -11,9 +11,9 @@
 
 
 if sys.maxsize >= 2**62:
-    dimensions = ((0, 0), (2, 2), (3, 3), (10, 10), (30, 30), (50, 50), (60, 60))
+    dimensions = ((1, 1), (2, 2), (3, 3), (10, 10), (30, 30), (50, 50), (60, 60))
```
Comment (malb) [inline tests/test_gso.py:14]: I think we should check these, those can happen as corner cases of higher-level algorithms

Comment (ludopulles) [reply tests/test_gso.py]: Fair enough, I have changed matrix generation in this test file to use `bits=max(1, m)` in that case, as `bits=0` gave a Segmentation Fault.
